### PR TITLE
[BEAM-151] Add support for RunnableOnService validation tests to runners module

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Default implementation for Dataflow requires you to specify the runnableOnServicePipelineOptions
system property with a valid project/runner/... to be able to execute on Google Cloud Dataflow.

Default implementation for Flink and Spark are to have it disabled.

Note the whitespace formatting fix on spark pom.xml

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @davorbonaci 
CC: @amitsela 
CC: @mxm 

Davor, after merge, please update Jenkins build:
mvn -pl runners/google-cloud-dataflow-java -am clean verify -DrunnableOnServicePipelineOptions=...

There -DrunnableOnServicePipelineOptions replaces -DdataflowOptions

File: runners/pom.xml [hunk 1]
```diff
@@ -39,4 +39,35 @@
     <module>spark</module>
   </modules>
 
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>2.18.1</version>
```
Comment (davorbonaci) [inline runners/pom.xml]: Version shouldn't be needed.

File: runners/google-cloud-dataflow-java/pom.xml [hunk 2]
```diff
@@ -41,21 +38,35 @@
...
     <timestamp>${maven.build.timestamp}</timestamp>
     <maven.build.timestamp.format>yyyy-MM-dd HH:mm</maven.build.timestamp.format>
-    <dataflow>com.google.cloud.dataflow</dataflow>
-    <runIntegrationTestOnService>false</runIntegrationTestOnService>
-    <testParallelValue>none</testParallelValue>
-    <testGroups></testGroups>
-    <dataflowProjectName></dataflowProjectName>
   </properties>
 
+  <!-- Disable integration tests unless told configured by a test flag.
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/pom.xml]: This profile disables automatic execution of RunnableOnService integration tests on Google Cloud Dataflow service. However, the tests will be executed if `runnableOnServicePipelineOptions` is specified.

File: runners/pom.xml [hunk 3]
```diff
@@ -39,4 +39,35 @@
...
+          <executions>
+            <execution>
+              <id>runnable-on-service-tests</id>
+              <phase>integration-test</phase>
+              <goals>
+                <goal>test</goal>
+              </goals>
+              <configuration>
+                <groups>com.google.cloud.dataflow.sdk.testing.RunnableOnService</groups>
+                <parallel>all</parallel>
```
Comment (davorbonaci) [inline runners/pom.xml]: used to be `both`, not `all`. Just confirming it is intentional.

**Comment by davorbonaci** [comment]
LGTM

File: runners/pom.xml (hunk 3 above)
Comment (lukecwik) [inline runners/pom.xml]: Yes, intentional. both has been deprecated

**Comment by lukecwik** [comment]
@davorbonaci Did you perform the cleanups on merge?

**Comment by davorbonaci** [comment]
Yup.

**Pull request closed without merging**